frontend/src/components/Pages/FrontPage/RobotCards/RobotStatusView.tsx

```diff
@@ -40,10 +44,26 @@ export function RobotStatusSection({ refreshInterval }: RefreshProps) {
         })
         //.catch((e) => handleError(e))
     }
-
-    var robotDisplay = robots.map(function (robot) {
-        return <RobotStatusCard key={robot.id} robot={robot} />
+    var filteredRobots = robots.filter(function (robot) {
+        return (
+            robot.robotAsset === ActiveAsset() ||
+            (typeof robot.robotAsset === 'string' && robot.robotAsset.includes('default')) ||
+            robot.robotAsset === undefined
+        )
     })
+
+    var robotDisplay
+
+    if (ActiveAsset() === '') {
+        robotDisplay = robots.map(function (robot) {
+            return <RobotStatusCard key={robot.id} robot={robot} />
+        })
+    } else {
+        robotDisplay = filteredRobots.map(function (robot) {
+            return <RobotStatusCard key={robot.id} robot={robot} />
+        })
+    }
```

Comment:
Have you tested that the robots change when you change the asset?

Reply:
Yes